libvast/vast/polymorphic_visitor.hpp

```diff
@@ -88,7 +88,7 @@ class lambda_visitor : public polymorphic_visitor<Result> {
 
   caf::optional<Result> do_visit(const void* ptr,
                                  const std::type_info& info) override {
-    result_t result;
+    caf::optional<result_t> result;
     if ((try_visit<Ts>(result, ptr, info) || ...))
```

Comment:
There's no longer a reason for this branch. We can always return `result` here, because `optional<void>` und `optional<unit>` are compatible.

Reply:
What do you mean by "branch" in this context?

Reply:
The `if` branch.

Reply:
Ah, right. Seems like you've fixed this in your branch already.